flink-dist/src/main/assemblies/plugin.xml

```diff
@@ -0,0 +1,80 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<assembly
+		xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.0"
+		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.0 http://maven.apache.org/xsd/assembly-1.1.0.xsd">
+	<id>plugin</id>
+	<formats>
+		<format>dir</format>
+	</formats>
+
+	<includeBaseDirectory>true</includeBaseDirectory>
+	<baseDirectory>flink-${project.version}</baseDirectory>
+
+	<files>
+		<!-- hadoop -->
+		<file>
+			<source>../flink-filesystems/flink-hadoop-fs/target/flink-hadoop-fs-${project.version}.jar</source>
+			<outputDirectory>plugin/flink-hadoop-fs</outputDirectory>
```

Comment:
Does the plugin system enforce a single directory for each plugin?

Reply:
It would be good to have all filesystems plugins be bundled in a `filesystems` directory imo, otherwise this will get messy in the long run.

Reply:
@zentol I think what you mean is that we should adopt `three-layer` dir structure right? About this question, I have a discussion with @StefanRRichter in the [JIRA ticket](https://issues.apache.org/jira/browse/FLINK-11955?focusedCommentId=16818826&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16818826). Eventually, we agreed `two layers` structures just like the presto's plugin style.